lesson3

Relaxiy opened the pull request.

Relaxiy committed:
lesson3

Relaxiy committed:
package lesson3;

import java.util.Scanner;

public class Task1 {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        int[] a = new int[n];
        a[0] = 1;
        a[1] = 1;
        System.out.println(a[0]);
        System.out.println(a[1]);

        for(int i = 2; i < a.length; i++) {
            a[i] = a[i - 1] + a[i - 2];
            System.out.println(a[i]);
        }

    }
}

Relaxiy committed:
package lesson3;

import java.util.Scanner;

public class Task1 {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        int[] a = new int[n];
        a[0] = 1;
        a[1] = 1;
        System.out.println(a[0]);
        System.out.println(a[1]);

        for(int i = 2; i < a.length; i++) {
            a[i] = a[i - 1] + a[i - 2];
            System.out.println(a[i]);
        }

    }
}

damalikov commented on src/lesson3/Task1.java:
```diff
@@ -0,0 +1,21 @@
+    public static void main(String[] args) {
+        Scanner in = new Scanner(System.in);
+        int n = in.nextInt();
+        int[] a = new int[n];
```
Имеет ли смысл создавать массив. Так как у нас не стоит задача сохранять элементы, а просто вывести их на экран.

damalikov commented on src/lesson3/Task2.java:
```diff
@@ -0,0 +1,51 @@
+
+public class Task2 {
+    public static void main(String[] args) {
+        while(true) {
```
Опять условие в цикле while не хорошее.

damalikov commented on src/lesson3/Task2.java:
```diff
@@ -0,0 +1,51 @@
+                        }
+                }
+                break;
+            }
```
Снова break

damalikov commented on src/lesson3/Task3.java:
```diff
@@ -0,0 +1,21 @@
+            if (inA.hasNextInt() && inB.hasNextInt()) {
+                int a = inA.nextInt();
+                int b = inB.nextInt();
+                while(a < b - 1) {
```
Я бы рекомендовал вывести последовательость через цикл for

damalikov commented on src/lesson3/Task7.java:
```diff
@@ -0,0 +1,32 @@
+                }
+            }
+
+            System.out.println();
```
а зачем нужен этот sout

damalikov commented on src/lesson3/Task9.java:
```diff
@@ -0,0 +1,19 @@
+        int value = rand.nextInt(4);
+        Scanner in = new Scanner(System.in);
+
+        int YourVal;
```
переменные с маленькой буквы

damalikov commented on src/lesson3/Task11.java:
```diff
@@ -0,0 +1,41 @@
+        int indexValue = 0;
+        Scanner in = new Scanner(System.in);
+        index = in.nextInt();
+        if (index <= mass.length - 1) {
```
Плохой стиль написание. Код очен плохо читается, когда происходит такой большой уровень вложенности 
if () {
     for () {
           if () {
               for() {
Что-то очень страшное и нечитаемое. 
Во -первых зачем первый массив , который доходит до введеного индекса.
Во-вторых снова while с break.
Попробуй просто достать элементы из первого массива минуя один единсвтенный и положить их всех во второй)

Relaxiy committed:
package lesson3;

import java.util.Scanner;

public class Task1 {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        int[] a = new int[n];
        a[0] = 1;
        a[1] = 1;
        System.out.println(a[0]);
        System.out.println(a[1]);

        for(int i = 2; i < a.length; i++) {
            a[i] = a[i - 1] + a[i - 2];
            System.out.println(a[i]);
        }

    }
}

Relaxiy committed:
package lesson3;

import java.util.Scanner;

public class Task1 {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        int[] a = new int[n];
        a[0] = 1;
        a[1] = 1;
        System.out.println(a[0]);
        System.out.println(a[1]);

        for(int i = 2; i < a.length; i++) {
            a[i] = a[i - 1] + a[i - 2];
            System.out.println(a[i]);
        }

    }
}
